cluster-operator/src/main/java/io/strimzi/operator/cluster/operator/resource/KafkaRoller.java

```diff
@@ -455,7 +455,9 @@ private RestartPlan restartPlan(int podId, Pod pod, RestartContext restartContex
                 "PodScheduled".equals(ps.getType())
                         && "Unschedulable".equals(ps.getReason())
                         && "False".equals(ps.getStatus()));
-        if (podStuck && !reasonToRestartPod.contains("Pod has old generation")) {
+        if (podStuck
+                && !reasonToRestartPod.contains("Pod has old generation")   // "Pod has old generation" is used with StatefulSets
+                && !reasonToRestartPod.contains("Pod has old revision")) {  // "Pod has old revision" is used with PodSets
```

Comment:
If I got the PR right. This is the only change needed for fixing #6203, right? All the other changes are for improving how handling changes on volumes for rolling pods.

Reply:
No, that is not the case. Both changes are needed. The rolling update for adding / removing volumes is before the regular rolling update. When it runs, it fails on the pending pods since it doesn't roll the pod to fix resource configuration, just to add or remove storage. So adding the check if this it really needed, helps to get to the regular rolling update which fixes the pending pods. So they are both part of the fix.

Reply:
Ok, thanks.